lib/java/com/google/android/material/slider/BaseSlider.java

```diff
@@ -2035,20 +2039,30 @@ private void invalidateTrack() {
     activeTicksPaint.setStrokeWidth(trackHeight / 2.0f);
   }
 
-  /**
-   * If this returns true, we can't start dragging the Slider immediately when we receive a {@link
-   * MotionEvent#ACTION_DOWN}. Instead, we must wait for a {@link MotionEvent#ACTION_MOVE}. Copied
-   * from hidden method of {@link View} isInScrollingContainer.
-   *
-   * @return true if any of this View's parents is a scrolling View.
-   */
   private boolean isInScrollingContainer() {
-    ViewParent p = getParent();
-    while (p instanceof ViewGroup) {
-      if (((ViewGroup) p).shouldDelayChildPressedState()) {
+    return isInVerticalScrollingContainer() || isInHorizontalScrollingContainer();
+  }
+
+  private boolean isInVerticalScrollingContainer() {
+    ViewParent parent = getParent();
+    while (parent instanceof View) {
+      if (((View) parent).canScrollVertically(1) ||
```

Comment:
Maybe here and in isInHorizontalScrollingContainer() we should also check shouldDelayChildPressedState(). Even if the parent could scroll, if shouldDelayChildPressedState() is false then I think that it should behave like it isn't in a scrolling container

Reply:
In fact, maybe it's good enough to just remove `isInScrollingContainer()` and use `isInVerticalScrollingContainer()` with that logic. The other call to `isInScrollingContainer()` in `ACTION_DOWN` should also be `isInVerticalScrollingContainer()`. No point in waiting until `ACTION_MOVE` if we're just going to capture the touch event anyways.

So maybe remove `isInScrollingContainer()` and `isInHorizontalScrollingContainer()` and just use `isInVerticalScrollingContainer()` with a check for `shouldDelayChildPressedState()`